### PR TITLE
docs: improve GitHub Integration PR section

### PR DESCRIPTION
**Pull request by levlaz**
This commit shows an example and clarifies the disctinction between using the `/merge` ref and the `/head` ref. This is based on the feedback we got from Mark here: https://github.com/dagger/dagger/pull/8194#discussion_r1724578389

### CODE REVIEW

File: docs/current_docs/integrations/github.mdx [hunk 1]
```diff
@@ -72,20 +72,36 @@ Replace `${{ secrets.SSH_PRIVATE_KEY }}` with your provider secret containing th
 
 By default, the Dagger `Directory` type works with both local directories and [remote Git repositories](../../cookbook#copy-a-directory-or-remote-repository-to-a-container). This makes it easy to work with the directory tree at the root of a Git repository or a given branch.
 
-Given a Dagger Function called `foo` that accepts a `Directory` as argument, you can pass it a GitHub pull request URL as argument like this:
+Given a Dagger Function called `foo` that accepts a `Directory` as argument. You can pass it a GitHub pull request URL.
```
Comment (vikram-dagger) [inline docs/current_docs/integrations/github.mdx:75]:
```suggestion
Given a Dagger Function called `foo` that accepts a `Directory` as argument, you can pass it a GitHub pull request URL. Two options are available:
```

File: docs/current_docs/integrations/github.mdx [hunk 2]
```diff
@@ -72,20 +72,36 @@ Replace `${{ secrets.SSH_PRIVATE_KEY }}` with your provider secret containing th
...
+
+You can use the `/merge` ref to pass a directory that has the head branch merged into the base branch like this:
```
Comment (vikram-dagger) [inline docs/current_docs/integrations/github.mdx:77]: Maybe reformat these as two bullets?

File: docs/current_docs/integrations/github.mdx [hunk 3]
```diff
@@ -72,20 +72,36 @@ Replace `${{ secrets.SSH_PRIVATE_KEY }}` with your provider secret containing th
...
+
+```shell
+dagger call foo --directory=https://github.com/ORGANIZATION/REPOSITORY#pull/NUMBER/head
+```
+
+
 #### Pull requests as modules
 
-If your GitHub repository contains a Dagger module, you can test the functionality of a specific branch by calling the Dagger module with the corresponding pull request URL, as shown below:
+If your GitHub repository contains a Dagger module, you can test the functionality of a specific branch by calling the Dagger module with the corresponding pull request URL.
```
Comment (vikram-dagger) [inline docs/current_docs/integrations/github.mdx:92]:
```suggestion
If your GitHub repository contains a Dagger module, you can test the functionality of a specific branch by calling the Dagger module with the corresponding pull request URL. Two options are available:
```

File: docs/current_docs/integrations/github.mdx [hunk 4]
```diff
@@ -72,20 +72,36 @@ Replace `${{ secrets.SSH_PRIVATE_KEY }}` with your provider secret containing th
...
 
+You can use the `/merge` ref to pass a directory that has the head branch merged into the base branch like this:
```
Comment (vikram-dagger) [inline docs/current_docs/integrations/github.mdx:94]: As above, maybe reformat as two bullets.

File: docs/current_docs/integrations/github.mdx (hunk 2 above)
Comment (levlaz) [reply docs/current_docs/integrations/github.mdx]: done

File: docs/current_docs/integrations/github.mdx (hunk 4 above)
Comment (levlaz) [reply docs/current_docs/integrations/github.mdx]: done

File: dagger.json [hunk 5]
```diff
@@ -2,6 +2,10 @@
   "name": "dagger-dev",
   "sdk": "go",
   "dependencies": [
+    {
+      "name": "gha",
```
Comment (marcosnils) [inline dagger.json:6]: :thinking: seems like a mistake?

Comment (vikram-dagger) [reply dagger.json:6]: @levlaz could you please check this comment?

**Pull request closed without merging**